addons/controllers/clusterbootstrap_controller.go

```diff
@@ -342,6 +349,18 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecret(cluster *
 		}
 	}
 
+	// Add cluster and package labels to secrets if not already present
+	// This helps us to track the secrets in the watch and trigger Reconcile requests when these secrets are updated
+	patchedSecret := *secret.DeepCopy()
```

Comment:
question: why do we need this pointer conversion ? Instead of `patchedSecret := secret.DeepCopy()` and `patchSecretWithLabels(patchedSecret, pkg.RefName, cluster.Name)` ?

Reply:
yes, please the extra conversion seems redundant.

Reply:
Fixed